### ISSUE TITLE
Compile error on 2019.3

### ISSUE THREAD

**Issue by epsmarkh** [adjust/unity_sdk#165]
The following line in AdjustEditor.cs is not compatible with 2019.3:

string xcodeTarget = xcodeProject.TargetGuidByName("Unity-iPhone"); 

It needs updating to:

#if UNITY_2019_3_OR_NEWER
            string xcodeTarget = xcodeProject.GetUnityMainTargetGuid();
#else
            string xcodeTarget = xcodeProject.TargetGuidByName("Unity-iPhone");
#endif

**Comment by uerceg** [comment]
Hi @epsmarkh 

Thanks for reporting this thing to us. We'll try to reproduce it and if we do, we'll have this solution in mind to put into first Unity SDK update after Unity IDE 2019.3 becomes official release. If this condition you wrote solves your issue, feel free to use it, but on our end we'll wait for 2019.3 to be official release and see if issue is still there. I fully understand that chances are low that this behaviour will change until official release, but we had few situations in the past where we were trying to anticipate some beta changes with SDK updates which at the end of the road ended up having completely different behaviour than in beta causing our SDK which tried to fix the issue to actually break other stuff.

Thank you one more time, we'll keep our eyes on this topic and keep you posted in here.

Cheers
Ugi

**Comment by uerceg** [comment]
Hi @epsmarkh 

Just wanted to update you that we're working on SDK update which will address this issue you mentioned (thanks for reporting one more time).

I am not sure if you actually tested your proposed solution inside of your app? From what we noticed in tests is that this actually solves the issue:

```csharp
#if UNITY_2019_3_OR_NEWER
            string xcodeTarget = xcodeProject.GetUnityFrameworkTargetGuid();
#else
            string xcodeTarget = xcodeProject.TargetGuidByName("Unity-iPhone");
#endif
```

So it's `xcodeProject.GetUnityFrameworkTargetGuid()` instead of `string xcodeTarget = xcodeProject.GetUnityMainTargetGuid()`. To be honest at this stage, still not quite sure why your suggested solution didn't work for us, since it beyond any doubt makes sense for frameworks we're adding during post build process and also `Other Linker Flags` to be part of app's target and not this newly introduced Unity framework target. But if we add this to app's target, we see that `-ObjC` flag we're adding to `Other Linker Flags` is being completely ignored while when we add it to newly introduced Unity framework target, it actually does the trick. Also what we noticed is that all the iOS frameworks which Unity was packing by default to app's target in earlier versions are now actually ending up inside of theirs `UnityFramework` target under `Link Binary With Libraries` section, so maybe this tells something in favour that Adjust stuff should be added in there as well.

We're planning on going with an update described above, but would still like to hear from you on whether it really worked for you when you instructed to `AdjustEditor` to perform changes in `GetUnityMainTargetGuid()` target.

Cheers

**Comment by uerceg** [comment]
Hey @epsmarkh.

We have just released [Unity SDK v4.20.1](https://github.com/adjust/unity_sdk/releases/tag/v4.20.1) update which should hopefully address this issue. In case you still encounter it, let us know.

Cheers
